src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -11904,12 +11922,12 @@ void Player_UpdateCommon(Player* this, PlayState* play, Input* input) {
         Player_ProcessControlStick(play, this);
 
         if (this->stateFlags1 & PLAYER_STATE1_27) {
-            D_808535E8 = 0.5f;
+            sWaterInfluence = 0.5f;
         } else {
-            D_808535E8 = 1.0f;
+            sWaterInfluence = 1.0f;
         }
 
-        D_808535EC = 1.0f / D_808535E8;
+        D_808535EC = 1.0f / sWaterInfluence;
```

Comment:
could call this `sInvWaterInfluence` while you're at it

Reply:
Could. It only gets used to influence how door cameras work. Unsure if we want to name it literally like this or after its single usage.

Reply:
I'm pretty sure its purpose is to optimize the divisions there

Reply:
can you elaborate

Reply:
It seems Link's "door opening" animation speed is multiplied by a factor of `sWaterInfluence`, so to compensate the timers in `Camera_ChangeDoorCam` have to be changed as well. Since they're units of time rather than speed you have to do division:
```c
Camera_ChangeDoorCam(Play_GetCamera(play, CAM_ID_MAIN), doorActor,
     play->transitionActors.list[GET_TRANSITION_ACTOR_INDEX(doorActor)]
         .sides[(doorDirection > 0) ? 0 : 1]
         .bgCamIndex,
     0, 38.0f / sWaterInfluence, 26.0f / sWaterInfluence, 10.0f / sWaterInfluence);
```
but divisions are slower than multiplication, so instead they created `sInvWaterInfluence = 1.0f / sWaterInfluence` and multiplied by that instead.

Maybe `sWaterInfluence` -> `sWaterSpeedFactor` or something would make it clearer here? The "speed" gives a hint to the units involved

Reply:
it wasnt clear to me how it was an optimization, but I guess its simply because its 1 division instead of 3.

I avoided "speed" purposefully because I didnt want to give the impression that it was only "speed" in the xzSpeed sense. It modifies xzSpeed, animation speed, turn speed, and door camera duration as well (as seen here). But if it would be more clear than I can probably get used to it.

Reply:
maybe "rate"?

Reply:
like `sWaterRateFactor`, `sWaterSpeedRateFactor`? both sound like a mouthful to me.
I think `sWaterSpeedFactor` sounds clean enough, I can just clarify my worry in a comment.

Reply:
Yeah, and you've already said `Influences different speed values.` in the comment anyway

Reply:
done